src/org/opensolaris/opengrok/web/stats/Statistics.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2016, 2018, Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.web.stats;
+
+import java.io.Reader;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.gson.Gson;
+import org.opensolaris.opengrok.web.SearchHelper;
+
+/**
+ * Framework for statistics gathering. So far used only by the webapp.
+ *
+ * @author Krystof Tulinger
+ */
+public class Statistics {
+
+    private static final Gson gson = new Gson();
+
+    private Instant timeStart = Instant.now();
+
+    private Map<String, Long> requestCategories = new TreeMap<>();
+    private Map<String, Long> timing = new TreeMap<>();
+    private Map<String, Long> timingMin = new TreeMap<>();
+    private Map<String, Long> timingMax = new TreeMap<>();
+    private long[] dayHistogram = new long[24];
+    private long[] monthHistogram = new long[31];
+    private long requests = 0;
+    private long minutes = 1;
+    private long requestsPerMinute = 0;
+    private long requestsPerMinuteMin = Long.MAX_VALUE;
+    private long requestsPerMinuteMax = Long.MIN_VALUE;
+
+    private AtomicLong searchRequests = new AtomicLong();
```

Comment:
I think this is doubling the value in `requestCategories['search']`

Reply:
Yes, it is. I missed `addRequest` call in `addRequestTime`...